tests/Microsoft.Identity.Test.Core.UIAutomation/MobileTestHelper.cs

```diff
@@ -74,8 +77,14 @@ public void AcquireTokenSilentTestHelper(ITestController controller, LabResponse
             AcquireTokenInteractiveHelper(controller, labResponse, CoreUiTestConstants.UiBehaviorLogin);
             VerifyResult(controller);
 
-            //acquire token for 2nd resource with refresh token
-            SetInputData(controller, labResponse.AppId, CoreUiTestConstants.DefaultScope, CoreUiTestConstants.UiBehaviorLogin);
+            controller.Tap(CoreUiTestConstants.SelectUser, XamarinSelector.ByAutomationId);
```

Comment:
At some point, we should stop using a test app with a complex UI and adopt a simpler strategy, similar to how mobile unit tests exist: 

- the test app has a drop-down that shows a list of test names and a button to run the test
- the automation iterates through this drop-down and runs each test
- all configuration is in memory, rather than having a zillion of input boxes, switches, tabs and buttons. 

We can even take a step further and use the Xamarin.Test infrastructure for running unit tests, where the test logic is injected from the test to the app and the test runner handles exceptions, failures etc. 

@MarkZuber @henrik-me @trwalke - thoughts?

Reply:
We had the idea to do something like this a while back but we never got around to doing it. The problem we have now is that we are using apps designed for customers to test out MSAL rather than apps designed specifically for testing. One great example of how to do this is with our old desktop automation app. This app had a handful of buttons to initiate different scenarios and if you click on one of these buttons, you were able to pass in a json string with all of the configuration properties for that scenario. This brought the complexity of that app way down as we only needed to focus on automating the web interactions rather than configuring the app with a series of buttons.

an example of what the configuration looked like:
{"authority":"https://login.microsoftonline.com/common","client_id":"3c1e0e0d-b742-45ba-a35e-01c664e14b16","redirect_uri":"urn:ietf:wg:oauth:2.0:oob","resource":"00000003-0000-0ff1-ce00-000000000000","prompt_behavior":"Always"}

Reply:
My $.02: The less UI we have to automate, the better.  And a button or drop list item per test makes it clear what the test breakdowns being executed.

Reply:
I like the idea of having a drop down. Can look into that at a later date, but not part of this PR.

---
In reply to: [296304990](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1226#discussion_r296304990) [](ancestors = 296304990)

Reply:
@jennyf19  -- agree with "not this pr".  but forward thinking, simplifying our ui automation makes sense.

Reply:
definitely.

---
In reply to: [296309507](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1226#discussion_r296309507) [](ancestors = 296309507)